DirectProgramming/DPC++FPGA/Tutorials/GettingStarted/fpga_template/src/fpga_template.cpp

```diff
@@ -0,0 +1,78 @@
+#include <iostream>
+
+// oneAPI headers
+#include <CL/sycl.hpp>
+#include <sycl/ext/intel/fpga_extensions.hpp>
+
+using namespace sycl;
+
+// Forward declare the kernel name in the global scope. This is an FPGA best
+// practice that reduces name mangling in the optimization reports.
+class SimpleVAdd;
+
+class SimpleVAddKernel {
+ public:
+  int *A_in;
+  int *B_in;
+  int *C_out;
+  int len;
+
+  void operator()() const {
+    for (int idx = 0; idx < len; idx++) {
+      int a_val = A_in[idx];
+      int b_val = B_in[idx];
+      int sum = a_val + b_val;
+      C_out[idx] = sum;
+    }
+  }
+};
+
+#define VECT_SIZE 256
+
+int main() {
+#if FPGA_SIMULATOR
+  std::cout << "using FPGA Simulator." << std::endl;
+  return sycl::ext::intel::fpga_simulator_selector{};
+#elif FPGA_HARDWARE
+  std::cout << "using FPGA Hardware." << std::endl;
+  return sycl::ext::intel::fpga_selector{};
+#else  // #if FPGA_EMULATOR
+  std::cout << "using FPGA Emulator." << std::endl;
+  return sycl::ext::intel::fpga_emulator_selector{};
+#endif
+
+  int count = VECT_SIZE;  // pass array size by value
+
+  // declare arrays and fill them
+  // allocate in shared memory so the kernel can see them
+  int *A = malloc_shared<int>(count, q);
+  int *B = malloc_shared<int>(count, q);
+  int *C = malloc_shared<int>(count, q);
+  for (int i = 0; i < count; i++) {
+    A[i] = i;
+    B[i] = (count - i);
+  }
+
```

Comment:
I think we should use the try catch mechanism for now, until we figure out if it may be useful or not.
If it is not useful, we'll remove it from all samples, if it is useful, then you'll need it anyway :)

Reply:
removing it as discussed in other calls

Reply:
put it back